tamr_unify_client/mastering/project.py

```diff
@@ -141,18 +142,39 @@ def published_cluster_versions(self, cluster_ids):
         """Retrieves version information for the specified published clusters.
 
         :param cluster_ids: The persistent IDs of the clusters to get version information for.
-        :type cluster_ids: list[str]
+        :type cluster_ids: iterable[str]
         :return: A stream of the published clusters.
         :rtype: Python generator yielding :class:`~tamr_unify_client.mastering.published_cluster.resource.PublishedCluster`
         """
-        stringified_ids = "\n".join(
-            json.dumps(cluster_id) for cluster_id in cluster_ids
-        )
-        url = self.api_path + "/publishedClusterVersions"
+        path = self.api_path + "/publishedClusterVersions"
+        return self._cluster_versions(cluster_ids, path, PublishedCluster)
+
+    def record_published_cluster_versions(self, record_ids):
+        """Retrieves version information for the published clusters of the given records.
+
+        :param record_ids: The Tamr IDs of the records to get cluster version information for.
+        :type record_ids: iterable[str]
+        :return: A stream of the relevant published clusters.
+        :rtype: Python generator yielding :class:`~tamr_unify_client.mastering.published_cluster.record.RecordPublishedCluster`
+        """
+        path = self.api_path + "/recordPublishedClusterVersions"
+        return self._cluster_versions(record_ids, path, RecordPublishedCluster)
+
+    def _cluster_versions(self, ids, endpoint, cluster_class):
```

Comment:
Calling convention within unify-client-python is to provide the class as the first parameter in these situations - e.g. `BaseResource.from_data(cls, ...)`